src/align.py

```diff
@@ -50,10 +50,10 @@ def make_modified_TarInfo(archive, target_dir=''):
 
 
 def get_flagstats(input_path, output_path):
-    command = shlex.split(
-        'samtools flagstat {infile}'.format(infile=input_path))
-    with open(output_path, 'wt') as f:
-        subprocess.call(command, stdout=f)
+    command = 'samtools flagstat {infile}'.format(infile=input_path)
+    output = subprocess.getoutput(command) + '\n'
```

Comment:
getoutput is an old method inherited from an old module ... subprocess.check_output was the preferred way (with shlex to avoid shell injection problems) and has good exception handling .... subprocess.run is the >= Python 3.5 way.  I know you had a specific reason to change this, so maybe this is the way you want to do it.  But this sort of thing gets done in so many places in pipeline code you might want to consider having one preferred way that you use consistently.  We can talk more about it if you have questions.

Reply:
Just curious, why not `subprocess.Popen` ?

Reply:
strattan, I'll change that one. vsoch, there is no difference there. As you surely know, call is just a wrapper around Popen, so under the hood that gets used. The fact that the convenience wrapper call blocks, whereas you can use Popen in a non blocking way makes no difference here. If you are not familiar with call, see CPython implementation here: https://hg.python.org/cpython/file/e0df7db13d55/Lib/subprocess.py#l459 Long story short: It is a matter of convenience to use call instead of Popen here.

Reply:
Thanks, understood! This is generally a hard call to make (I made a pun!) I've found that there is no one solution to fit all use cases, and generally have to vary the stdout (and other) variables to get the functionality that I want (either processing and sending back output line by line vs just showing the output). If it helps, here is an example function I've found works in most cases --> https://github.com/singularityhub/singularity-cli/blob/master/spython/utils/terminal.py#L128 

the "capture" variable is most important - if I start a service I turn it off, but for anything to capture output I set it to True. And then the regular expression is a hack for not making the progress bar display on like, 1000 different lines :P